TrainingExtensions/torch/src/python/aimet_torch/tensor_quantizer.py

```diff
@@ -612,28 +610,36 @@ def _per_channel_quantize_dequantize(tensor_to_quantize_dequantize: torch.Tensor
         :param channel_axis: Axis along which per channel quantize dequantize is performed
         :return: quantized dequantized tensor
         """
-        quantized_tensors = []
-        for i, minimum in enumerate(encoding_min):
-            tensor_slice = tensor_to_quantize_dequantize.select(channel_axis, i).contiguous(memory_format=torch.contiguous_format)
-            tensor = torch.clamp(tensor_slice, minimum.item(), encoding_max[i].item())
-            tensor = torch.round(tensor / delta[i].item()) + offset[i].item()
-            tensor = (tensor - offset[i].item()) * delta[i].item()
-            quantized_tensors.append(tensor)
-        quantized_tensor = torch.stack(tuple(quantized_tensors), dim=channel_axis)
+        if len(tensor_to_quantize_dequantize.shape) > 1:
+            encoding_min = grad_fn.broadcast_to_tensor(tensor_to_quantize_dequantize, encoding_min, channel_axis)
+            encoding_max = grad_fn.broadcast_to_tensor(tensor_to_quantize_dequantize, encoding_max, channel_axis)
+            delta = grad_fn.broadcast_to_tensor(tensor_to_quantize_dequantize, delta, channel_axis)
+            offset = grad_fn.broadcast_to_tensor(tensor_to_quantize_dequantize, offset, channel_axis)
 
-        return quantized_tensor
+        tensor = torch.clamp(tensor_to_quantize_dequantize, encoding_min, encoding_max)
+        tensor = torch.round(tensor / delta) + offset
+        tensor = (tensor - offset) * delta
+
+        return tensor
 
     @staticmethod
     def backward(ctx, grad):
         # Retrieve saved tensors for gradient calculations
-        tensor, encoding_min, encoding_max, n, p, bitwidth = ctx.saved_tensors
+        tensor, encoding_min, encoding_max, n, p, bitwidth, channel_axis = ctx.saved_tensors
+        channel_axis = channel_axis.item()
 
         scale = (encoding_max - encoding_min) / (2 ** bitwidth.data - 1)
         offset = torch.round(-encoding_min / scale)
 
-        tensor_grad = ParameterQuantizer.compute_dloss_by_dx(tensor, grad, scale, offset, n, p)
-        tensor_encoding_max_grad = ParameterQuantizer.compute_dloss_by_dmax(tensor, grad, scale, offset, n, p)
-        tensor_encoding_min_grad = ParameterQuantizer.compute_dloss_by_dmin_using_dmax(tensor_encoding_max_grad)
+        if len(scale) > 1 and len(tensor.shape) > 1:
```

Comment:
Copying my comment from the other PR: 

Do we need two if checks here? If len(scale) > 1, would that mean we are in per channel mode and thus offset would be expected to be > 1 as well?

Perhaps better would be to have one if check, inside which we have an assert statement that len(offset) > 1, and take care of both scale and offset broadcasting in the if check. As it currently stands, the code would not hiccup here if len(scale) > 1 while offset is not, even though it seems like that would be an invalid case.

And do we ever expect len of scale or offset to be > 1 but tensor.shape len not > 1? If not, this should also be an assert check instead of an additional condition in the if statement.